src/index.tsx

```diff
@@ -144,8 +151,153 @@ function App() {
+    }
+    if (selectionMode === "selected") {
+      setSelectionMode(false);
+      setSelectionStart(undefined);
+      setSelectionEnd(undefined);
+    }
+  }
+
+  function onMapClick(info: PickingInfo) {
+    if (!selectionMode || selectionMode === "selected") return;
+    console.log("onclick info", info);
+    if (selectionEnd !== undefined) {
+      setSelectionStart(undefined);
+      setSelectionEnd(undefined);
+    } else if (selectionStart !== undefined && selectionEnd === undefined) {
+      setSelectionEnd([[info.x, info.y], info.coordinate]);
+      const width = Math.abs(info.x - selectionStart[0][0]);
+      const height = Math.abs(info.y - selectionStart[0][1]);
+      const left = Math.min(selectionStart[0][0], info.x);
+      const top = Math.min(selectionStart[0][1], info.y);
+      const selectedObjects = mapRef.current?.pickObjects({
+        x: left,
+        y: top,
+        width,
+        height,
+      });
+      setSelectionMode("selected");
+      setHoverBBoxLayer(false);
+      console.log("selected on map", selectedObjects);
+      setSelectionObjectCount(selectedObjects ? selectedObjects.length : 0);
+    } else {
+      setSelectionStart([[info.x, info.y], info.coordinate]);
+      setSelectionEnd(undefined);
+    }
+  }
+
+  function onMapHover(hoverInfo: PickingInfo) {
+    if (selectionMode !== "selecting") return;
+    const hoverCoords = hoverInfo.coordinate;
+    if (selectionStart && hoverCoords) {
+      const pt1 = selectionStart[1];
+      const pt2 = hoverCoords;
+      if (!pt1 || !pt2) return;
+      const data = [
+        {
+          polygon: [pt1, [pt1[0], pt2[1]], pt2, [pt2[0], pt1[1]], pt1],
+        },
+      ];
+      const bboxLayer = new PolygonLayer({
+        id: "selection-layer",
+        data,
+        filled: true,
+        getFillColor: [0, 0, 0, 50],
+        stroked: true,
+        getLineWidth: 2,
+        lineWidthUnits: "pixels",
+        getPolygon: (d) => d.polygon,
+      });
+      console.log(bboxLayer);
+      setHoverBBoxLayer(bboxLayer);
+    }
+    return;
+  }
+
+  const selectionIndicator = useMemo(() => {
+    if (!selectionMode) return undefined;
+    if (selectionStart && selectionEnd) {
+      console.log(`Map coords: ${selectionStart[1]} ${selectionEnd[1]}`);
+      const pt1 = selectionStart[1];
+      const pt2 = selectionEnd[1];
+      if (!pt1 || !pt2) return undefined;
+      const data = [
+        {
+          polygon: [pt1, [pt1[0], pt2[1]], pt2, [pt2[0], pt1[1]], pt1],
+        },
+      ];
+      console.log("selection data", data);
+      return new PolygonLayer({
+        id: "selection-layer",
+        data,
+        filled: true,
+        getFillColor: [0, 0, 0, 30],
+        stroked: true,
+        getLineWidth: 2,
+        lineWidthUnits: "pixels",
+        getPolygon: (d) => d.polygon,
+      });
+    } else if (selectionStart) {
+      // Show the selection start point (note this does not show the proposed bounding box, but could be done)
```

Comment:
```suggestion
      // Show the selection start point
```
Since a hover box is now shown